### PR TITLE
♻️ Initial Refactor commit [APP-687]

### PR DESCRIPTION
**Pull request by bainternet**
Added new code base with namespace helper classes and utils.

### CODE REVIEW

File: .github/scripts/update-version-in-files.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+				},
+				{
+					from:  /= NEXT_VERSION =/m,
+					to: `= ${ VERSION } =`,
+				},
+			]
+		);
+
+		// update version in image-optimization.php
+		await replaceInFileWithArray( './image-optimization.php',
```
Comment (nirbhayel) [inline .github/scripts/update-version-in-files.js:43]: Looks like file referred is incorrect

File: .github/scripts/update-version-in-files.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			[
+				{
+					from:  /\* Version: (.*)/m,
+					to: `* Version: ${ VERSION }`,
+				},
+				{
+					from:  /define\( \'IMAGE_OPTIMIZER_VERSION\', \'(.*)\' \)\;/m,
+					to: `define( 'IMAGE_OPTIMIZER_VERSION', '${ VERSION }' );`,
```
Comment (nirbhayel) [inline .github/scripts/update-version-in-files.js:51]: incorrect constant name

File: tests/phpunit/helpers/test-base.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?php
+namespace SiteMailer\Tests\Helpers;
```
Comment (nirbhayel) [inline tests/phpunit/helpers/test-base.php:2]: incorrect namespace

File: .github/scripts/update-version-in-files.js (hunk 1 above)
Comment (nirbhayel) [inline .github/scripts/update-version-in-files.js:43]: incorrect file name

**Review by nirbhayel — COMMENTED** [review]
Please check comments

File: classes/utils/assets.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+	/**
+	 * get_assets_version
+	 *
+	 * @param string $version
+	 *
+	 * @return string
+	 */
+	private static function get_assets_version( string $version = '' ) : string {
+		return empty( $version ) ? \SITE_MAILER_VERSION : $version;
```
Comment (pkniazevych) [inline classes/utils/assets.php:59]: Let's to update the name

File: classes/utils/assets.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	 * @param string $asset_name
+	 * @param string $asset_type
+	 * @param string $suffix
+	 *
+	 * @return string
+	 */
+	private static function get_assets_path( string $asset_name, string $asset_type, string $suffix = '' ) : string {
+		return \SITE_MAILER_ASSETS_URL . '/build/' . $asset_name . $suffix . '.' . $asset_type;
+		//TODO: Check if we need this
+		//return \SITE_MAILER_ASSETS_URL . $asset_type . '/' . ( '' === $suffix ? 'dev/' : '' ) . $asset_name . $suffix . '.' . $asset_type;
```
Comment (pkniazevych) [inline classes/utils/assets.php:96]:
```suggestion
```

File: classes/utils/assets.php [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+	/**
+	 * enqueue_app_assets
+	 *
+	 * @param string $handle
+	 * @param bool $with_css
+	 */
+	public static function enqueue_app_assets( string $handle = '', bool $with_css = true ) : void {
+		$dir = \SITE_MAILER_ASSETS_PATH . 'build/';
+		$url = \SITE_MAILER_ASSETS_URL . 'build/';
```
Comment (pkniazevych) [inline classes/utils/assets.php:107]: Let's update those also

File: classes/utils/assets.php [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+		// add translation support
+		wp_set_script_translations( $handle, 'elementor-sw' );
+
+		if ( ! $with_css ) {
+			return;
+		}
+		// enqueue css
+		$css_file_name = 'style-' . $handle . '.css';
+		$css_version = ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? filemtime( $dir . $css_file_name ) : \SITE_MAILER_VERSION;
```
Comment (pkniazevych) [inline classes/utils/assets.php:134]: And here

File: ruleset.xml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress.Elementor">
+	<description>Elementor Coding Standard</description>
+
+	<arg name="parallel" value="8" />
+
+	<config name="text_domain" value="site-mailer" />
```
Comment (pkniazevych) [inline ruleset.xml:7]: Let's update the text domain

File: classes/utils/assets.php [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+	 * get_assets_path
+	 *
+	 * @param string $asset_name
+	 * @param string $asset_type
+	 * @param string $suffix
+	 *
+	 * @return string
+	 */
+	private static function get_assets_path( string $asset_name, string $asset_type, string $suffix = '' ) : string {
+		return \SITE_MAILER_ASSETS_URL . '/build/' . $asset_name . $suffix . '.' . $asset_type;
```
Comment (pkniazevych) [inline classes/utils/assets.php:94]: Let's update `SITE_MAILER_ASSETS_URL`

**Review by pkniazevych — CHANGES_REQUESTED** [review]
Please, check the comments.

File: classes/logger.php [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+	private static function log( string $log_level, $message ): void {
+		$backtrace = debug_backtrace(); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_debug_backtrace
+
+		$class    = $backtrace[2]['class'] ?? null;
+		$type     = $backtrace[2]['type'] ?? null;
+		$function = $backtrace[2]['function'];
+
+		if ( $class ) {
+			$message = '[Site Mailer]: ' . $log_level . ' in ' . "$class$type$function()" . ': ' . $message;
```
Comment (DinetsV) [inline classes/logger.php:22]:
```suggestion
			$message = '[One Click Accessibility]: ' . $log_level . ' in ' . "$class$type$function()" . ': ' . $message;
```

File: classes/logger.php [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+		} else {
+			$message = '[Site Mailer]: ' . $log_level . ' in ' . "$function()" . ': ' . $message;
```
Comment (DinetsV) [inline classes/logger.php:24]:
```suggestion
			$message = '[One Click Accessibility]: ' . $log_level . ' in ' . "$function()" . ': ' . $message;
```

**Review by DinetsV — APPROVED** [review]

**Pull request merged**